sdk/tables/Azure.Data.Tables/src/Queryable/ExpressionNormalizer.cs

```diff
@@ -134,6 +142,11 @@ internal Expression VisitMethodCallNoRewrite(MethodCallExpression call)
                 return CreateCompareExpression(visited.Arguments[0], visited.Arguments[1]);
             }
 
+            if (visited.Method == ReflectionUtil.DictionaryGetItemMethodInfo && visited.Arguments.Count == 1 && visited.Arguments[0] is ConstantExpression ce)
```

Comment:
Looking at ExpressionNormalizer and ExpressionWriter I wonder if we really need both or if they can be merged into one with additional benefit of avoiding creating of more Expression nodes just to throw them away.

Reply:
In some cases the normalizer is simplifying the expression to be written by the writer. I'd like to do an optimization pass at some point to address issues like this and others.